src/components/Search/Search.js

```diff
@@ -246,18 +274,59 @@ const Search = () => {
       dispatch(setClickResults([]))
     })
 
-    // remove clicked footprint highlight
-    if (clickedFootprintHighlightRef.current)
-      clickedFootprintHighlightRef.current.clearLayers()
-
-    // remove image layer
-    if (clickedFootprintImageLayerRef.current)
-      clickedFootprintImageLayerRef.current.clearLayers()
+    // if user does not click on the same image, clear layers
+    if (!currentImageClickedRef.current) {
+      // only remove layers if the user clicked on a different footprint
+      if (clickedFootprintImageLayerRef.current)
+        clickedFootprintImageLayerRef.current.clearLayers()
+      if (clickedFootprintHighlightRef.current)
+        clickedFootprintHighlightRef.current.clearLayers()
+    }
 
     // remove existing footprints from map
-    resultFootprintsRef.current.eachLayer(function (layer) {
-      resultFootprintsRef.current.removeLayer(layer)
-    })
+    resultFootprintsRef.current.clearLayers()
+  }
+
+  const getQueryVal = () => ({ 'eo:cloud_cover': { gte: 0, lte: _cloudCover } })
+
+  async function fetchAPIitems() {
+    // build datetime input
+    const combinedDateRange = convertDateForURL(dateTimeRef.current)
+
+    // get viewport bounds and setup bbox parameter
+    const bbox = setupCommaSeparatedBbox(map)
+
+    const searchParams = new Map([
+      ['bbox', bbox],
+      ['datetime', combinedDateRange],
+      ['collections', selectedCollectionRef.current],
+      ['limit', API_MAX_ITEMS]
+    ])
+
+    if (showCloudSliderRef.current)
+      searchParams.set(
+        'query',
+        encodeURIComponent(JSON.stringify(getQueryVal()))
+      )
+
+    const searchParamsStr = [...searchParams]
+      .reduce((obj, x) => {
+        obj.push(x.join('='))
+        return obj
+      }, [])
+      .join('&')
+
+    dispatch(setSearchParameters(searchParamsStr))
+
+    const searchURL = `${process.env.REACT_APP_STAC_API_URL}/search?${searchParamsStr}`
+
+    const response = await fetch(searchURL)
+    if (!response.ok) {
+      const message = `An error has occurred: ${response.status}`
+      throw new Error(message)
```

Comment:
```suggestion
      throw new Error(`An error has occurred: ${response.status}`)
```

I usually only put the error message into a variable if I need to do two things with it, like log and use it in the Error msg.

Reply:
Completed in next PR